include/eve/module/core/regular/impl/sum_of_prod.hpp

```diff
@@ -0,0 +1,41 @@
+//==================================================================================================
+/*
+  EVE - Expressive Vector Engine
+  Copyright : EVE Contributors & Maintainers
+  SPDX-License-Identifier: MIT
+*/
+//==================================================================================================
+#pragma once
+
+#include <eve/concept/value.hpp>
+#include <eve/detail/apply_over.hpp>
+#include <eve/detail/implementation.hpp>
+
+namespace eve::detail
+{
+  template<floating_real_value T, floating_real_value U, floating_real_value V, floating_real_value W >
+  EVE_FORCEINLINE auto sum_of_prod_(EVE_SUPPORTS(cpu_)
+                                   , const T& a
+                                   , const U& b
+                                   , const V& c
+                                   , const W& d
+                                   ) noexcept
+  {
+    return arithmetic_call(sum_of_prod, a, b, c, d);
+  }
+
+  template<floating_real_value T>
+  EVE_FORCEINLINE auto sum_of_prod_(EVE_SUPPORTS(cpu_)
+                                   , const T& a
+                                   , const T& b
+                                   , const T& c
+                                   , const T& d
+                                   ) noexcept
+  requires(has_native_abi_v<T>)
```

Comment:
Again, aggregated

Reply:
Have to check, that's old jtlap's code

Reply:
OK, the aggregated case is handled by the previous overload.
We don't want to call aggregated directly on multi-operators expression, so we call the arithmetic_call skeleton instead.
Now, we usually do that with a if constexpr in all the other code so it requires to be streamlined

Reply:
Well, nope, the other overload takes care of both aggregation AND multi types variant.
So no this code is correct and it pass the aggregation tests already

Reply:
Never used the "arithmetic_call" if it does what's needed, OK.
Removing overloads would be nice I think

Reply:
The multi-type cases is needed to handle wide/scalar mixture.
I would say to let it this way for now